### ISSUE TITLE
PriorityQueue needs rework

### ISSUE THREAD

**Issue by rajatjain1997** [aimacode/aima-python#1024]
As pointed out by #919 and #941, the implementation of A* needs rework.

**Comment by zeph1yr** [comment]
Is this issue resolved? If resolved, why is it still open?

**Comment by rajatjain1997** [comment]
Yes. My PR works to resolve this issue and those I've mentioned in it. I think they'll be closed once the PR is merged.

**Comment by dmeoli** [comment]
This has been addressed. The current `PriorityQueue` (utils.py) now implements exactly the disambiguation proposed here:
- `x in pq` (`__contains__`) → whether the item is present, **regardless** of its f-value;
- `pq[x]` (`__getitem__`) → the item's f-value, raising `KeyError` if absent (dict-like).

It also pushes `(f(item), counter, item)` with a **monotonic counter** tie-breaker, so equal-priority items are never compared against each other — which fixes the non-deterministic / revisit behaviour behind #919 and #941 (both since closed) and makes the queue safe for non-comparable payloads. Closing as resolved.
